addons/mrp/models/stock_warehouse.py

```diff
@@ -170,9 +170,21 @@ def _get_locations_values(self, vals):
         values = super(StockWarehouse, self)._get_locations_values(vals)
         def_values = self.default_get(['manufacture_steps'])
         manufacture_steps = vals.get('manufacture_steps', def_values['manufacture_steps'])
+        code = vals.get('code') or self.code
+        code = code.replace(' ', '').upper()
         values.update({
-            'pbm_loc_id': {'name': _('Pre-Production'), 'active': manufacture_steps in ('pbm', 'pbm_sam'), 'usage': 'internal'},
-            'sam_loc_id': {'name': _('Post-Production'), 'active': manufacture_steps == 'pbm_sam', 'usage': 'internal'},
+            'pbm_loc_id': {
+                'name': _('Pre-Production'),
+                'active': manufacture_steps in ('pbm', 'pbm_sam'),
+                'usage': 'internal',
+                'barcode': code + '-PREPRODUCTION'
+            },
+            'sam_loc_id': {
+                'name': _('Post-Production'),
+                'active': manufacture_steps == 'pbm_sam',
+                'usage': 'internal',
+                'barcode': code + '-POSTPRODUCTION'
```

Comment:
@Whenrow perhaps nice that "PREPRODUCTION" is also translatable?

Reply:
post